File: src/ds-visuals/stack/application/inToPost/code/stack.py
class Stack(object):
	"""docstring for Stack"""
	def __init__(self):
		super(Stack, self).__init__()
		self.top = -1
		self.MAX = 10
		self.stack = ['']*self.MAX

	def push(self,element):
		if(self.top == self.MAX-1):
			print("Error- stack full")
		else:
			self.top += 1
			self.stack[self.top] = element
		return;

	def pop(self):
		if(self.top == -1):
			print("Error- stack empty")
		else:
			self.top -= 1
			return self.stack[self.top+1]
		return -1

	def peek(self):
		return self.stack[self.top]

	def display(self):
		print('stack: ', end="")
		for i in range(self.top+1):
			print(str(self.stack[i])+" ", end="")
		print()

	def createStack(self, *elements):
		for element in elements:
			self.push(element)

File: src/ds-visuals/stack/application/inToPost/code/stack.java
class Stack{
	final int MAX = 10;
	char stack[] = new char[MAX];
	int top = -1;

	Stack(char...arg){
		this.createStack(arg);
	}

	void push(char element){
		if(this.top == MAX-1)
			System.out.println("Error- stack full");
		else
			this.stack[++(this.top)] = element;
		return;
	}

	char pop(){
		if(this.top == -1)
			System.out.println("Error- stack empty");
		else
			return this.stack[(this.top)--];
		return '\n';
	}

	char peek(){
		return this.stack[top];
	}

	void createStack(char... arg){
		for(int i = 0; i < arg.length; i++)
			this.push(arg[i]);
		return;
	}

	void display(){
	    System.out.printf("stack: ");
		for(int i = 0; i <= this.top; i++)
			System.out.printf(this.stack[i]+" ");
		System.out.println();
	}
}
